aries_vcx/src/indy/ledger/transactions.rs

```diff
@@ -497,29 +499,50 @@ pub async fn get_attr(pool_handle: PoolHandle, did: &str, attr_name: &str) -> Vc
     libindy_submit_request(pool_handle, &get_attrib_req).await
 }
 
+
+
 pub async fn get_service(pool_handle: PoolHandle, did: &Did) -> VcxResult<AriesService> {
     let did_raw = did.to_string();
-    let  did_raw = match  did_raw.rsplit_once(':'){
-        None => {did_raw}
-        Some((_,value)) => {value.to_string()}
+    let did_raw = match did_raw.rsplit_once(':') {
+        None => { did_raw }
+        Some((_, value)) => { value.to_string() }
     };
-    let attr_resp = get_attr(pool_handle, &did_raw, "service").await?;
+    let attr_resp = get_attr(pool_handle, &did_raw, "endpoint").await?;
     let data = get_data_from_response(&attr_resp)?;
-    let ser_service = match data["service"].as_str() {
-        Some(ser_service) => ser_service.to_string(),
-        None => {
-            warn!("Failed converting service read from ledger {:?} to string, falling back to new single-serialized format", data["service"]);
-            data["service"].to_string()
-        }
-    };
-    serde_json::from_str(&ser_service).map_err(|err| {
-        VcxError::from_msg(
-            VcxErrorKind::SerializationError,
-            format!("Failed to deserialize service read from the ledger: {:?}", err),
-        )
-    })
+    if !data["endpoint"].is_null() {
+
+        let endpoint_json = data["endpoint"].to_string();
+        let endpoint: EndpointService = serde_json::from_str(&endpoint_json)?;
+        let _recipient = if endpoint.recipient_keys.is_none() {
+            let verkey = get_verkey_from_ledger(pool_handle, &did_raw).await.unwrap();
+            vec![verkey]
+        } else {
+            endpoint.recipient_keys.unwrap()
+        };
+        return Ok(AriesService::create()
+            .set_recipient_keys(_recipient)
+            .set_service_endpoint(endpoint.endpoint)
+            .set_routing_keys(endpoint.routing_keys.unwrap()))
+    }else {
```

Comment:
formatting, missing space before `else`

Reply:
There's number of formatting issues, check if your IDE support some auto-formatting feature. 
Alternatively it's also possible to reformat individual files with `cargo fmt` or `rustfmt`  - beware, I strongly recommend commit all changes before playing with `cargo fmt` or `rustfmt`, otherwise it's easy to mix up functional changes